### PR TITLE
Introduce a clear everything button in multiselect and search autocompletes

### PR DESCRIPTION
**Pull request by jmuscireum**
## Breaking

This is breaking, as the new CSS class name (`token-autocomplete-delete-button`) has to be used in overriding CSS rules and in custom token renderers.

Replace `token-autocomplete-token-delete` with `token-autocomplete-delete-button`.
CSS rules might have to be adjusted.

### Description
This fixes some small issues (see the first two commits) and introduces a clear button in multiselect and search autocompletes to clear all tokens and user input (remaining commits)

**Correct height for empty readonly selects:**
Bildschirmfoto 2024-06-18 um 16 08 09

**Turn the arrow up when the suggestions are displayed:**
Bildschirmfoto 2024-06-18 um 16 08 18

**New symbol for optional single selects:**
Bildschirmfoto 2024-06-18 um 16 08 31

**Multiselect with clear button:**
Bildschirmfoto 2024-06-18 um 16 08 36

### Additional Notes

- This PR fixes or works on following ticket(s): [SE-13734](https://scireum.myjetbrains.com/youtrack/issue/SE-13734)
- This PR is related to PR: scireum/sirius-web/pull/1434

### Checklist

- [x] Code change has been tested and works locally
- [x] TSC (TypeScript Compiler) has been run and changes are reflected in the dist folder

### CODE REVIEW

**Review by fhaScireum — APPROVED** [review]

File: lib/token-autocomplete.ts [hunk 1]
```diff
@@ -1092,6 +1093,7 @@ class TokenAutocomplete {
          * Shows the suggestions dropdown to the user.
          */
         showSuggestions() {
+            this.container.classList.add('select-arrow-active');
```
Comment (sabieber) [inline lib/token-autocomplete.ts:1131]: **Optional:** I would prefer if the class name would also be prefixed with `token-autocomplete-`

File: lib/token-autocomplete.ts [hunk 2]
```diff
@@ -773,8 +773,8 @@ class TokenAutocomplete {
             this.parent.textInput.textContent = tokenText;
             this.parent.textInput.contentEditable = 'false';
             this.parent.textInput.blur();
-            if (this.options.optional && tokenText !== '') {
-                this.container.classList.add('optional-singleselect-with-value');
+            if (tokenText !== '') {
+                this.container.classList.add('has-value');
```
Comment (sabieber) [inline lib/token-autocomplete.ts:813]: **Optional:** same as above

**Review by sabieber — APPROVED** [review]

File: lib/token-autocomplete.ts [hunk 3]
```diff
@@ -472,6 +480,20 @@ class TokenAutocomplete {
...
+                me.updateHasValue();
+            });
+
+            if (this.options.showClearButton) {
+                parent.container.querySelector('.token-autocomplete-delete-button')?.addEventListener('click', function () {
+                    me.clear(true);
+                    me.clearCurrentInput();
+                    me.updateHasValue();
+
+                    me.container.dispatchEvent(new CustomEvent('cleared'));
```
Comment (ymo-sci) [inline lib/token-autocomplete.ts:494]: Maybe rename to input-cleared or tokens-cleared?

**Review by ymo-sci — APPROVED** [review]

**Review by mko-sci — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by sabieber — COMMENTED** [review]
Thanks a lot 👌🏻

**Pull request merged**